Mathlib/CategoryTheory/Types/Monomorphisms.lean

```diff
@@ -0,0 +1,128 @@
+/-
+Copyright (c) 2025 Joël Riou. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Joël Riou
+-/
+import Mathlib.CategoryTheory.Limits.Connected
+import Mathlib.CategoryTheory.Limits.Types.Filtered
+import Mathlib.CategoryTheory.MorphismProperty.Limits
+import Mathlib.CategoryTheory.MorphismProperty.TransfiniteComposition
+
+/-!
+# Stability properties of monomorphisms in `Type`
+
+In this file, we show that in the category `Type u`, monomorphisms
+are stable under cobase change, filtered colimits and
+transfinite compositions.
+(The stability by retracts holds in any category: it is shown
+in the file `CategoryTheory.MorphismProperty.Retract`.)
+
+-/
+
+universe v' u' u
+
+namespace CategoryTheory
+
+open MorphismProperty Limits Types
+
+namespace Types
+
+instance : (monomorphisms (Type u)).IsStableUnderCobaseChange where
+  of_isPushout {X₁ X₂ X₃ X₄ t l r b} sq ht := by
+    simp only [monomorphisms.iff, mono_iff_injective] at ht ⊢
+    exact Limits.Types.pushoutCocone_inr_injective_of_isColimit sq.flip.isColimit ht
+
+lemma isStableUnderColimitsOfShape_monomorphisms_of_isFiltered
+    (J : Type u') [Category.{v'} J] [IsFiltered J] :
+    (monomorphisms (Type u)).IsStableUnderColimitsOfShape J := by
```

Comment:
How come `IsStableUnderColimitsOfShape` is not a class?

Reply:
It should have been done some time ago... I do it in #24773.